lib/chef/util/powershell/ps_credential.rb

```diff
@@ -0,0 +1,32 @@
+#
+# Author:: Jay Mundrawala (<jdm@chef.io>)
+#
+# Copyright:: Copyright (c) 2015 Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'chef/win32/crypto'
+
+class Chef::Util::Powershell
+  class PSCredential
+    def initialize(username, password)
+      @username = username
+      @encrypted_password = Chef::ReservedNames::Win32::Crypto.encrypt(password)
+    end
+
+    def to_psobject
+      "New-Object System.Management.Automation.PSCredential('#{@username}',\('#{@encrypted_password}' | ConvertTo-SecureString))"
```

Comment:
https://technet.microsoft.com/en-us/library/bb726984.aspx
Logon names can't contain certain characters. Invalid characters are
" / \ [ ] : ; | = , + * ? < >
Logon names can contain all other special characters, including spaces, periods, dashes, and underscores.

Example:  "O'Reilly"